datasets/coco2014/metadata.json

```diff
@@ -68,147 +69,138 @@
+      "contentUrl": "http://images.cocodataset.org/annotations/image_info_test2014.zip",
+      "encodingFormat": "application/zip",
+      "sha256": ""
     },
     {
       "@type": "sc:FileSet",
       "name": "imageinfo",
-      "containedIn": [
-        "#{image_info_test2014.zip}"
-      ],
-      "includes": "annotations/image_info_test.json",
-      "encodingFormat": "application/json"
+      "containedIn": "#{image_info_test2014.zip}",
+      "encodingFormat": "application/json",
+      "includes": "annotations/image_info_test.json"
     }
   ],
   "recordSet": [
     {
       "@type": "ml:RecordSet",
       "name": "split_enums",
       "description": "Maps split names to semantic values.",
-      "source": "#{splits.csv}",
-      "key": "#{name}",
       "field": [
         {
+          "@type": "ml:Field",
           "name": "name",
           "description": "One of: train, val, test.",
-          "@type": "ml:Field",
-          "source": "#{splits.csv/name}",
-          "dataType": "sc:Text"
+          "dataType": "sc:Text",
+          "source": "#{splits.csv/name}"
         },
         {
+          "@type": "ml:Field",
           "name": "url",
           "description": "Corresponding mlcommons.org definition URL",
-          "@type": "ml:Field",
-          "source": "#{splits.csv/url}",
           "dataType": [
             "sc:Url",
             "wd:Q3985153"
-          ]
+          ],
+          "source": "#{splits.csv/url}"
         }
-      ]
+      ],
+      "key": "#{name}",
+      "source": "#{splits.csv}"
     },
     {
-      "name": "images",
       "@type": "ml:RecordSet",
-      "key": "#{img_id}",
+      "name": "images",
       "field": [
         {
-          "name": "image_filename",
           "@type": "ml:Field",
+          "name": "image_filename",
           "description": "The filename of the image. eg: COCO_train2014_000000000003.jpg",
           "dataType": "sc:Text",
           "source": "#{image-files/filename}"
         },
         {
-          "name": "image_content",
           "@type": "ml:Field",
+          "name": "image_content",
           "description": "The content of the image.",
           "dataType": "sc:ImageObject",
           "source": "#{image-files/content}"
         },
         {
-          "name": "split",
           "@type": "ml:Field",
+          "name": "split",
           "dataType": [
             "sc:Text",
             "wd:Q3985153"
           ],
+          "references": "#{split_enums}",
           "source": {
-            "data": "#{image-files/fullpath}",
             "applyTransform": {
               "regex": "^(train|val|test)2014.zip/.+2014/.*\\.jpg$"
-            }
-          },
-          "references": "#{split_enums}"
+            },
+            "data": "#{image-files/fullpath}"
+          }
         }
-      ]
+      ],
+      "key": "#{img_id}"
     },
     {
-      "name": "captions",
       "@type": "ml:RecordSet",
-      "key": "#{id}",
```

Comment:
I feel like it would be clearer to have the key above field (field at the bottom). Your call.